init.sh

```diff
@@ -18,15 +18,29 @@ echo "Updating working directory in workflow files..."
 sed -i "s|working-directory:\s*'\.'|working-directory: '$install_directory'|g" "$temp_directory/sp-files/.github/workflows"/*
 echo "Updating .readthedocs.yaml configuration..."
 sed -i "s|configuration:\s*sp-docs/conf\.py|configuration: $install_directory/conf.py|g" "$temp_directory/sp-files/.readthedocs.yaml"
-sed -i "s|requirements:\s*sp-docs/\.sphinx/requirements\.txt|requirements: $install_directory/.sphinx/requirements.txt|g" "$temp_directory/.readthedocs.yaml"
+sed -i "s|requirements:\s*sp-docs/\.sphinx/requirements\.txt|requirements: $install_directory/.sphinx/requirements.txt|g" "$temp_directory/sp-files/.readthedocs.yaml"
 
 # Create the specified installation directory if it doesn't exist
 if [ ! -d "$install_directory" ]; then
     echo "Creating the installation directory: $install_directory"
     mkdir -p "$install_directory"
 fi
 
-# Copy the contents of the starter pack repository to the installation directory
+# Check if .gitignore exists in the destination directory
+# If exists, append the contents of the source .gitignore to the destination .gitignore
+if [ -f "$install_directory/.gitignore" ]; then
+    cat "$temp_directory/sp-files/.gitignore" >> "$install_directory/.gitignore"
+    rm "$temp_directory/sp-files/.gitignore"
+fi
+
+# Check if Makefile exists in the destination directory
+# If exists, append the contents of the source Makefile to the destination Makefile
+if [ -f "$install_directory/Makefile" ]; then
+    cat "$temp_directory/sp-files/Makefile" >> "$install_directory/Makefile"
```

Comment:
I think we should output a warning here in addition - because this might not work out of the box (for example, if the Makefile already has some of the target names). So telling the user to check the Makefile before running it would be helpful.

Reply:
In most cases, concatenating Makefiles would create more confusion. I'd suggest either grafully failing (with a descriptive message), or creating a back up if an existing Makefile is found in the dest. dir.

Reply:
Thanks for the feedback @ru-fu @rkratky ! I changed it to a user prompt in [a5f6382](https://github.com/canonical/sphinx-docs-starter-pack/pull/256/commits/a5f63821aba9b5fc65d412db534f661b123a437f) - users can choose to create the docs Makefile as a backup `Makefile.backup.{date}`, or skip this step